internal/model/puller.go

```diff
@@ -422,6 +417,20 @@ nextFile:
 
 // handleDir creates or updates the given directory
 func (p *Puller) handleDir(file protocol.FileInfo) {
+	var err error
+	events.Default.Log(events.ItemStarted, map[string]interface{}{
+		"folder":  p.folder,
+		"item":    file.Name,
+		"details": db.ToTruncated(file),
+	})
+	defer func() {
```

Comment:
No, as that would evaluate the value of err upon deferring.
One of the gotcha's which I've slipped on before.